src/pydvl/value/shapley/montecarlo.py

```diff
@@ -152,6 +149,117 @@ def permutation_montecarlo_shapley(
     return map_reduce_job()
 
 
+def permutation_montecarlo_classwise_shapley(
+    u: Utility,
+    label: int,
+    *,
+    done: StoppingCriterion,
+    truncation: TruncationPolicy,
+) -> ValuationResult:
+    """
+    Samples a random subset of the complement set and computes the truncated monte carlo
+    estimator.
+
+    :param u: Utility object with model, data, and scoring function. The scoring
+        function has to be of type :class:`~pydvl.utils.score.ClassWiseScorer`.
+    :param done: Function checking whether computation must stop.
+    :param label: The label for which the complement set shall be sampled.
+    :param truncation: A callable which decides whether to interrupt processing a
+        permutation and set all subsequent marginals to zero.
+    :return: The updated result object.
+    """
+
+    algorithm_name = "classwise_shapley"
+    result = ValuationResult.zeros(
+        algorithm=algorithm_name,
+    )
+
+    _, y_train = u.data.get_training_data(u.data.indices)
+    class_indices_set, class_complement_indices_set = _split_into_index_set_by_label(
+        u.data.indices,
+        y_train,
+        label,
+    )
+    _, complement_y_train = u.data.get_training_data(class_complement_indices_set)
+    indices_permutation = np.random.permutation(class_indices_set)
+
+    for subset_idx, subset_complement in enumerate(
+        random_powerset_group_conditional(
+            class_complement_indices_set, complement_y_train
+        )
+    ):
+        logger.debug(f"Sampling set #{subset_idx} ")
+        result += _permutation_montecarlo_shapley_rollout(
+            u,
+            indices_permutation,
+            conditional_indices=subset_complement,
+            truncation=truncation,
+            algorithm_name=algorithm_name,
+        )
+        if done(result):
+            break
+
+    return result
+
+
+def _permutation_montecarlo_shapley_rollout(
+    u: Utility,
+    permutation: NDArray[np.int_],
+    *,
+    truncation: TruncationPolicy,
+    algorithm_name: str,
+    conditional_indices: Optional[NDArray[np.int_]] = None,
+) -> ValuationResult:
+    """
+    A truncated version of a permutation-based MC estimator for class-wise shapley
+    values. It generates a permutation p[i] of the class label indices and iterates over
+    the subsets starting from the set containing only one element up to the full set of
```

Comment:
If this statement is true, then the function needs to be changed, as we discussed: the first marginal must be between the empty set and the set containing the point of interest (but yes, this is basically measuring the noise of the estimator with random parameters, because training on one point is meaningless, so it is up to discussion whether we should change this for all methods, see #346 )

Reply:
Done as discussed from first point. Now it matches description.